### PR TITLE
feat: add auth hooks config

### PR DESCRIPTION
**Pull request by J0**
## What kind of change does this PR introduce?

Add configuration for auth hooks so that developers can test their hooks locally. Key fields for each hook are:

Enabled - true/false
URI - this should be in the form `pg-functions://<database>/<schema>/<function-name>`

This was tested by locally building the cli `go build .` and then `../cli/init` + `../cli/start`. Check that Docker has `GOTRUE_HOOK_<HOOK_NAME>_ENABLED`

### CODE REVIEW

File: internal/utils/config.go [hunk 1]
```diff
@@ -347,6 +349,17 @@ type (
 		TestOTP             map[string]string `toml:"test_otp"`
 	}
 
+	hook struct {
+		MFAVerificationAttempt      extensibilityPointConfig `toml:"mfa_verification_attempt" mapstructure:"mfa_verification_attempt"`
+		PasswordVerificationAttempt extensibilityPointConfig `toml:"password_verification_attempt" mapstructure:"password_verification_attempt"`
+		CustomAccessToken           extensibilityPointConfig `toml:"custom_access_token" mapstructure:"custom_access_token"`
+	}
+
+	extensibilityPointConfig struct {
```
Comment (sweatybridge) [inline internal/utils/config.go:358]: I'm tempted to name this struct `hookConfig` for simplicity. Wdyt?

File: internal/utils/config.go [hunk 2]
```diff
@@ -311,6 +312,7 @@ type (
 		EnableRefreshTokenRotation bool `toml:"enable_refresh_token_rotation"`
 		RefreshTokenReuseInterval  uint `toml:"refresh_token_reuse_interval"`
 		EnableManualLinking        bool `toml:"enable_manual_linking"`
+		Hook                       hook `toml:"hook" mapstructure:"hook"`
```
Comment (sweatybridge) [inline internal/utils/config.go:315]:
```suggestion
		Hook                       hook `toml:"hook"`
```

File: internal/utils/config.go [hunk 3]
```diff
@@ -347,6 +349,17 @@ type (
 		TestOTP             map[string]string `toml:"test_otp"`
 	}
 
+	hook struct {
+		MFAVerificationAttempt      extensibilityPointConfig `toml:"mfa_verification_attempt" mapstructure:"mfa_verification_attempt"`
+		PasswordVerificationAttempt extensibilityPointConfig `toml:"password_verification_attempt" mapstructure:"password_verification_attempt"`
+		CustomAccessToken           extensibilityPointConfig `toml:"custom_access_token" mapstructure:"custom_access_token"`
```
Comment (sweatybridge) [inline internal/utils/config.go:355]:
```suggestion
		MFAVerificationAttempt      extensibilityPointConfig `toml:"mfa_verification_attempt"`
		PasswordVerificationAttempt extensibilityPointConfig `toml:"password_verification_attempt"`
		CustomAccessToken           extensibilityPointConfig `toml:"custom_access_token"`
```

No need to use mapstructure. See comment here https://github.com/supabase/cli/blob/main/internal/utils/config.go#L230

File: internal/utils/config.go (hunk 1 above)
Comment (J0) [reply internal/utils/config.go:358]: sounds good

File: internal/utils/templates/init_config.toml [hunk 4]
```diff
@@ -110,6 +110,11 @@ template = "Your code is {{ `{{ .Code }}` }} ."
 [auth.sms.test_otp]
 # 4152127777 = "123456"
 
+[auth.hook.custom_access_token]
```
Comment (sweatybridge) [inline internal/utils/templates/init_config.toml:113]: Can we also add a comment describing this field?

**Review by sweatybridge — APPROVED** [review]

File: internal/utils/templates/init_config.toml (hunk 4 above)
Comment (J0) [reply internal/utils/templates/init_config.toml:114]: Good catch, thanks!

**Pull request merged**